File: SIMINOV-CONNECT-TEMPLATE/src/siminov/connect/template/services/UnregisterDevice.java
package siminov.connect.template.services;

import siminov.connect.connection.design.IConnectionRequest;
import siminov.connect.connection.design.IConnectionResponse;
import siminov.connect.exception.ServiceException;
import siminov.connect.service.Service;

public class UnregisterDevice extends Service {

	private static final String SERVICE_NAME = "SIMINOV-CONNECT-NOTIFICATION-SERVICE";
	private static final String API_NAME = "UNREGISTER-DEVICE";
	
	public UnregisterDevice() {
		setService(SERVICE_NAME);
		setApi(API_NAME);
	}
	
	public void onStart() {

	}

	public void onQueue() {
		
	}

	public void onPause() {
		
	}

	public void onResume() {
		
	}

	public void onFinish() {
		
	}

	public void onApiInvoke(IConnectionRequest connectionRequest) {
		
	}

	public void onApiFinish(IConnectionResponse connectionResponse) {
		
	}

	public void onTerminate(ServiceException serviceException) {
		
	}

	public void onRestart() {
		
	}
}

File: hybrid-phonegap-sample/assets/www/Liquor/Services/UnregisterDevice.js
/** 
 * [SIMINOV FRAMEWORK]
 * Copyright [2015] [Siminov Software Solution LLP|support@siminov.com]
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 **/

if(window['document'] == undefined) {
    var Log = require('../../Siminov/Log/Log');
    var Function = require('../../Siminov/Function/Function');
    var Service = require('../../Siminov/Service/Service');
    
    module.exports = UnregisterDevice;    
}


function UnregisterDevice() {

    Service.apply(this, arguments);
    
	this.setService(UnregisterDevice.SERVICE_NAME);
	this.setRequest(UnregisterDevice.REQUEST_NAME);
	
	this.onStart = function() {

	}

	this.onQueue = function() {
		
	}

	this.onPause = function() {
		
	}

	this.onResume = function() {
		
	}

	this.onFinish = function() {
		
	}

	this.onRequestInvoke = function(connectionRequest) {
		
	}

	this.onRequestFinish = function(connectionResponse) {
		
	}

	this.onTerminate = function(serviceException) {
		
	}
}



UnregisterDevice.SERVICE_NAME = "SIMINOV-HYBRID-NOTIFICATION-SERVICE";
UnregisterDevice.REQUEST_NAME = "UNREGISTER-DEVICE";


Function.extend(Service, UnregisterDevice);
